### PR TITLE
Radix modal, now with drawer power

### PR DESCRIPTION
**Pull request by wingfeatherwave**
Now you can turn a `<Modal>` look like a right-sliding drawer with `<Modal drawer>`

(FYI, the create vault modal is still a regular ol modal, I was just using that as an example.)

Screen Shot 2022-09-22 at 11 43 07 AM

### CODE REVIEW

File: src/pages/VaultsPage/VaultsPage.tsx [hunk 1]
```diff
@@ -186,16 +186,18 @@ const VaultsPage = () => {
           )}
         </>
       )}
-      {modal === 'create' && currentOrg && (
+      {currentOrg && (
```
Comment (levity) [inline src/pages/VaultsPage/VaultsPage.tsx:189]: do the animations still work if you re-add `!!modal` to this? i think there's a slight performance advantage to not rendering the modal component at all until it's opened

Comment (levity) [reply src/pages/VaultsPage/VaultsPage.tsx:189]: ah, they don't, the close animation gets lost...

**Review by levity — APPROVED** [review]

**Comment by levity** [comment]
looks great! good call adding it as a prop to Modal

File: src/pages/VaultsPage/VaultsPage.tsx (hunk 1 above)
Comment (wingfeatherwave) [reply src/pages/VaultsPage/VaultsPage.tsx:189]: Radix doesn't add the modal to the dom until it's triggered, so there's no render hit before mount... but the radix root primitive I guess is still in the react tree somewhere.  But to answer your question the open animation will work, but not the close.

**Pull request merged**